x-pack/plugins/reporting/export_types/csv_from_savedobject/server/lib/generate_csv_search.ts

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore
+import { buildEsQuery } from '@kbn/es-query';
+import { Request } from 'hapi';
+import moment from 'moment';
+import { KbnServer, Logger } from '../../../../types';
+// @ts-ignore
+import { fieldFormatMapFactory } from '../../../csv/server/lib/field_format_map';
+// @ts-ignore
+import { createGenerateCsv } from '../../../csv/server/lib/generate_csv';
```

Comment:
@joelgriffith had a question on how this worked, and when looking into it, we started to wonder if this module creates a scroll context that should be manually cleared (opposed to cleared after a timeout)

Reply:
We do call clear scroll when the job is done:
```
server    log   [23:48:24.322] [debug][reporting][savedobject-csv] executing scroll request
server    log   [23:48:24.344] [debug][reporting][savedobject-csv] executing scroll request
server    log   [23:48:24.368] [debug][reporting][savedobject-csv] executing scroll request
server    log   [23:48:24.391] [debug][reporting][savedobject-csv] executing scroll request
server    log   [23:48:24.407] [debug][reporting][savedobject-csv] executing scroll request
server    log   [23:48:24.434] [warning][reporting][savedobject-csv] max Size Reached
server    log   [23:48:24.435] [debug][reporting][savedobject-csv] executing clearScroll request
```